File: OSharp/Interfaces/IInputSystem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OSharp.Interfaces
{
    public interface IInputSystem
    {
        void SetCursorPosition(O_Location point);
        void ShowCursor();
        void HideCursor();
    }
}

File: OnyxCLR/IInputSystem.cpp
#include "IInputSystem.h"
#include "InputSystem.h"

void IInputSystem::SetCursorPosition(OSharp::Interfaces::O_Location point)
{
    InputSystem::Get()->SetCursorPosition(Point(point.X, point.Y));
}

void IInputSystem::ShowCursor()
{
    InputSystem::Get()->ShowCursor();
}

void IInputSystem::HideCursor()
{
    InputSystem::Get()->HideCursor();
}
